build.gradle

```diff
@@ -8,32 +8,31 @@ plugins {
     id 'java'
     id 'idea'
     id 'application'
-    id 'com.diffplug.spotless' version '6.3.0'
-    id 'com.github.spotbugs' version '4.2.0'
+
+    id 'com.diffplug.spotless' version '6.9.1'
+    id 'com.github.spotbugs' version '5.0.9'
+    id 'org.gradle.test-retry' version '1.4.0'
+
     // Fix guava -jre vs -android dependency resolution.
     // The terra-cloud-resource library requires the -jre guava version. Without this plugin, we may
     // use the wrong guava version, leading to runtime errors.
     // See also https://blog.gradle.org/guava
-    id 'de.jjohannes.missing-metadata-guava' version '0.4'
-    id 'org.gradle.test-retry' version '1.4.0'
+    id 'de.jjohannes.missing-metadata-guava' version '31.1.1'
 }
 
-if (hasProperty('buildScan')) {
-    buildScan {
-        termsOfServiceUrl = 'https://gradle.com/terms-of-service'
-        termsOfServiceAgree = 'yes'
-    }
-}
+// If true, search local repository (~/.m2/repository/) first for dependencies.
+def useMavenLocal = false
 
-// The tools/publish-release.sh script depends on this version string being of the format "version = '1.2.3'"
```

Comment:
Curious why this moved to settings.gradle?

Reply:
Keeping this consistent structure across other repos. 
I will further update the build files in this PR itself --> keeping it more modular.